pkg/upstream/transformer.go

```diff
@@ -162,9 +162,23 @@ func (t *StaticRedirectTransform) Modify(r *Request) (*Request, error) {
 // - contrib.go.opencensus.io
 // - go.uber.org
 type GoGetTransform struct {
-	domains    map[string]bool // only implement redirect metadata
-	httpClient *http.Client
-	log        loggy.Logger
+	redirectAll bool
+	domains     map[string]bool // only implement redirect metadata
+	httpClient  *http.Client
+	log         loggy.Logger
+}
+
+// NewRedirectTransform creates a GoGetTransform where any module URI
+// will be redirected to wherever the go-get meta HTML tag in the domain
+// indicates.
+func NewRedirectTransform() Transform {
```

Comment:
this really should have GoGet in the name, because "redirect" is used in a lot of different contexts - maybe `NewAutoGoGetRedirectTransform` ?